.github/workflows/build-release.yaml

```diff
@@ -0,0 +1,200 @@
+          release-suffix: ${{ matrix.release-suffix }}
+          zksync-version: ${{ needs.setup.outputs.zksync-version }}
+          solc-version: ${{ needs.setup.outputs.solc-version }}
+
+  build-windows:
+    name: Windows
+    needs: setup
+    if: needs.setup.outputs.skip-windows == 'false'
+    runs-on: windows-2019-github-hosted-16core
+    steps:
+
+      - name: Checkout source
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Install LLVM and Clang
+        uses: KyleMayes/install-llvm-action@v1
+        with:
+          version: "15.0.7"
+
+      - name: Prepare environment
+        run: |
+          git config --global user.email "runner@github.com"
+          git fetch --all
+          # try to apply win patch
+          git cherry-pick 860226e25dca397b5afd70680530963712aff050 || git cherry-pick --abort
+          # latest llvm for MSVC
+          git clone -q --depth=1 --branch=main https://github.com/zufuliu/llvm-utils.git c:\projects\llvm-utils
+          Invoke-Item "c:\projects\llvm-utils\VS2017\install.bat"
+          clang --version
+
+
+      - name: Install windows Deps
+        run: |
+          if ( -not (Test-Path "deps\boost") ) {
+            New-Item -ItemType Directory -Force -Path "deps"
+
+            Invoke-WebRequest -URI "https://github.com/Kitware/CMake/releases/download/v3.27.1/cmake-3.27.1-windows-x86_64.zip" -OutFile cmake.zip
+
+            tar -xf cmake.zip
+            mv cmake-3.27.1-windows-x86_64 "deps\cmake"
+
+            # and serving it a page that requires JavaScript.
+            Invoke-WebRequest -URI "https://boostorg.jfrog.io/artifactory/main/release/1.77.0/source/boost_1_77_0.zip" -OutFile boost.zip
+
+            tar -xf boost.zip
+            cd boost_1_77_0
+            .\bootstrap.bat --with-toolset=clang
+            .\b2 -j4 -d0 toolset=clang cxxflags="-stdlib=libc++" linkflags="-stdlib=libc++" link=static runtime-link=static variant=release threading=multi address-model=64 --prefix="..\deps\boost" --with-filesystem --with-system --with-program_options --with-test --with-regex --with-thread install
+            if ( -not $? ) { throw "Error building boost." }
+            cd ..
+          }
+
+      - name: Building the Solidity compiler
+        run: |
+          $env:path += ";deps\cmake\bin\"
+          $env:CXXFLAGS="-Wno-narrowing -Qunused-arguments -Wno-everything -DBOOST_REGEX_NO_LIB -D_REGEX_MAX_STACK_COUNT=200000L -DJSON_USE_INT64_DOUBLE_CONVERSION -std=c++17 -stdlib=libc++"
+          mkdir build
+          cd build
+          $boost_dir=(Resolve-Path ..\deps\boost\lib\cmake\Boost-*)
+          cmake .. -D SOLC_VERSION_ZKSYNC="${{ needs.setup.outputs.zksync-version }}" -D USE_Z3="OFF" -DBoost_DIR="$boost_dir\" -DBoost_USE_STATIC_RUNTIME=1 -DBoost_USE_STATIC_LIBS=1 -DBoost_COMPILER=clang15 -T LLVM_v142 -DPEDANTIC=OFF -DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded
+          cmake --build . -j 10 --target install --config Release
```

Comment:
Can we use nproc/sysctl -n hw.ncpu here as well?

Reply:
Yes, IIRC we decided to remove this hardcode.

Reply:
To be honest, I'm afraid to touch this powershell part... we can try to use `[Environment]::ProcessorCount` here, but as I don't have access to this Windows host, I'm not 100% sure it will work. I prefer to keep it hardcoded 10 for Windows for now.

Reply:
Ah, if it's only for Windows, it's fine.
What strange Windows machine has 10 threads by the way? 😅

Reply:
Maybe change it to the actual number of threads of our runner.

Reply:
I'll ask DevOps to give me access to it, I don't have creds.

Reply:
Ok, so, it's our common runners, we're using `windows-2019-github-hosted-16core`... it contains 16 cores by default, so we can try to put 16.